yaml_interfaces.md

```diff
@@ -0,0 +1,164 @@
+# GeoIPS Yaml-based Plugin Options
+Internally, we would like Plugins to behave the same, regardless of whether they are Module-based or Yaml-based. This
+means that we would like to automatically discover Plugins of both types and automatically create Python objects for
+each Plugin. This will allow them to share functionality without needing to duplicate code.
+
+## Current Situation
+
+### Module-based Plugins
+At present, we have updated GeoIPS to automatically discover and objectify Module-based plugins. This is achieved by
+implementing a class for each GeoIPS Interface and a second class for Plugins associated with each interface. Classes
+include:
+- `BaseInterface` and `BaseInterfacePlugin`: Base classes which should not be instantiated directly.
+- `AlgorithmsInterface` and `AlgorithmsInterfacePlugin`: Classes for the algorithms interface and algorithm plugins.
+- `ReadersInterface` and `ReadersInterfacePlugin`
+- `ColorMapsInterface` and `ColorMapsInterfacePlugin`
+- etc...
+
+Plugins of these types are discovered by referencing entry points defined in a package's `setup.py`. These look like
+this:
+```python
+setup(
+    ...
+    entry_points={
+        'geoips.readers': [
+            'abi_netcdf=geoips.interface_modules.readers.abi_netcdf:abi_netcdf',
```

Comment:
@jsolbrig You are considering changing how this is set up when we switch to pyproject.toml, correct?  (And when we switch ffrom interface_modules to plugins subdirectory)?

Reply:
Yes, I'd like to quit pointing to the actual function and, instead, point to the module itself. I think that, for now, I'll keep the namespaces for different interfaces but may consider getting rid of them in the future, too. If we do that, everything would fall directly under the `geoips` namespace, but I think that is something for later.